pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -308,7 +310,34 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 		cdLog.Errorf("error getting job: %v", err)
 		return reconcile.Result{}, err
 	} else {
+		// If install job exists but the cd generation is different, recreate the job after deleting it
+		// provided the cd is not installed
 		cdLog.Infof("cluster job exists, successful: %v", cd.Status.Installed)
+		if !cd.Status.Installed {
+			cdGeneration := strconv.FormatInt(cd.Generation, 10)
+			cdLog.Infof("cluster generation: %v", cdGeneration)
+			if existingJob.Annotations != nil {
+				if generation, ok := existingJob.Annotations[clusterDeploymentGeneration]; ok {
+					if generation != cdGeneration {
+						err = r.Delete(context.TODO(), existingJob, client.PropagationPolicy(metav1.DeletePropagationForeground))
+						if err != nil {
+							cdLog.WithError(err).Errorf("error deleting outdated install job")
+							return reconcile.Result{}, err
+						} else {
+							cdLog.Infof("successfully deleted outdated install job")
+							err = r.Delete(context.TODO(), cfgMap, client.PropagationPolicy(metav1.DeletePropagationForeground))
```

Comment:
We should handle the configmap separately in the block where we check if it exists. If it already exists, we should update it to contain the values we expect.

Reply:
This one was my fault, I suggested that we handle the job and the install config map in lockstep, if we determine we need to delete the job, we know we should probably delete the configmap, and let both be regenerated on the re-sync.

Reply:
it's probably a good idea to always update the configmap if it exists, in case it's changed by other means.

Reply:
hmm, as I finished typing that, I realized that could be bad :) If we already have a job that's based on a current generation, then there's really no need to overwrite the configmap. We could put the generation annotation on the configmap as well though.

Reply:
Should I go ahead with the annotation on configmap?